src/trajectory/Interpolated.cpp

```diff
@@ -16,6 +16,13 @@ Interpolated::Interpolated(
   // Do nothing
 }
 
+//==============================================================================
+Interpolated::~Interpolated()
+{
+  for (auto waypoint : mWaypoints)
```

Comment:
Nit: to avoid unnecessary copy
```suggestion
  for (auto& waypoint : mWaypoints)
```